### ISSUE TITLE
Plugin crash with SQL Managed Instance data/resource

### ISSUE THREAD

**Issue by BenMurphy93** [hashicorp/terraform-provider-azurerm#21601]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.5 (also happens on 1.4.6 via pipeline)

### AzureRM Provider Version

3.54.0

### Affected Resource(s)/Data Source(s)

azurerm_mssql_managed_instance

### Terraform Configuration Files

```hcl
# SQL MI
data "azurerm_mssql_managed_instance" "sqlmi" {
  name                = "dev-sqlmi"
  resource_group_name = "dev-rg"
}

# Configure the Azure Provider
provider "azurerm" {
  features {}
  # Use Azure AD for Terraform accessing Storage Accounts
  storage_use_azuread = true
}

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3"
    }
  }
}
```

### Debug Output/Panic Output

```shell
data.azurerm_mssql_managed_instance.sqlmi: Reading...

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│ 
│   with data.azurerm_mssql_managed_instance.sqlmi,
│   on main.tf line 3, in data "azurerm_mssql_managed_instance" "sqlmi":
│    3: data "azurerm_mssql_managed_instance" "sqlmi" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v3.54.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x10906e530]

goroutine 131 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance.MsSqlManagedInstanceDataSource.Read.func1({0x10af3ba28, 0x14001e8e240}, {0x14002912000, {0x10af3c1d0, 0x14000611170}, 0x14002848600, 0x0, {0x10af3d350, 0x10f3ec808}})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/mssql_managed_instance_data_source.go:158 +0x50
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*DataSourceWrapper).DataSource.func2({0x10af3ba28, 0x14001e8e240}, 0x14001ed3328?, {0x10a38a780?, 0x14002912000?})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_data_source.go:48 +0xe8
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.diagnosticsWrapper.func1({0x10af3ba28, 0x14001e8e240}, 0x0?, {0x10a38a780, 0x14002912000})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:185 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x140008f0c40, {0x10af3ba60, 0x140022ccd50}, 0xd?, {0x10a38a780, 0x14002912000})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:724 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0x140008f0c40, {0x10af3ba60, 0x140022ccd50}, 0x14002848500, {0x10a38a780, 0x14002912000})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:943 +0x118
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0x1400129a360, {0x10af3ba60?, 0x140022ccc30?}, 0x1400287e1a0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1179 +0x304
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0x1400183b220, {0x10af3ba60?, 0x140022cc2a0?}, 0x1400216a0a0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:658 +0x368
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x10ab203e0?, 0x1400183b220}, {0x10af3ba60, 0x140022cc2a0}, 0x14000a72000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140004c4000, {0x10af4b3e0, 0x1400168b1e0}, 0x140011c8a20, 0x14001280cf0, 0x10f3ac290, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xb7c
google.golang.org/grpc.(*Server).handleStream(0x140004c4000, {0x10af4b3e0, 0x1400168b1e0}, 0x140011c8a20, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x290

Error: The terraform-provider-azurerm_v3.54.0_x5 plugin crashed!
```

### Expected Behaviour

SQL MI should have been read

### Actual Behaviour

Plugin crashes

### Steps to Reproduce

1. terraform init
2. terraform plan

### Important Factoids

Same thing happens when the SQLMI is referenced as a resource, issue not present in 3.53.0

**Comment by LaurentLesle** [comment]
Same with the sqlmi instance creation:

```
Stack trace from the terraform-provider-azurerm_v3.54.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x48ddf60]

goroutine 818 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance.MsSqlManagedInstanceResource.Create.func1({0x67361c8, 0x40027a3b00}, {0x4000517400, {0x6736970, 0x400061b4b8}, 0x4002c4d400, 0x0, {0x6737af0, 0xac44a68}})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go:267 +0x50
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func2({0x67361c8, 0x40027a3b00}, 0x0?, {0x52c60c0?, 0x4000517400?})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:52 +0xf4
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.diagnosticsWrapper.func1({0x67361c8, 0x40027a3b00}, 0x0?, {0x52c60c0, 0x4000517400})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:185 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x400086d340, {0x6736200, 0x40022fe5d0}, 0xd?, {0x52c60c0, 0x4000517400})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:707 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x400086d340, {0x6736200, 0x40022fe5d0}, 0x40007a6410, 0x4002c4d180, {0x52c60c0, 0x4000517400})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0x844
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x40010f22b8, {0x6736200?, 0x40022fe4b0?}, 0x4002000d20)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xbac
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x4001a00d20, {0x6736200?, 0x4001d0d800?}, 0x40018cf5e0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5a5b140?, 0x4001a00d20}, {0x6736200, 0x4001d0d800}, 0x40018cf570, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x16c
google.golang.org/grpc.(*Server).processUnaryRPC(0x40000de000, {0x6745bc0, 0x4001b0e1a0}, 0x400183ea20, 0x40014618f0, 0xac040a0, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xb68
google.golang.org/grpc.(*Server).handleStream(0x40000de000, {0x6745bc0, 0x4001b0e1a0}, 0x400183ea20, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x294

Error: The terraform-provider-azurerm_v3.54.0_x5 plugin crashed!
```

**Comment by jasonrberk** [comment]
confirmed downgrade to `=3.53.0` fixed the issue

**Comment by sschoof** [comment]
I think https://github.com/hashicorp/terraform-provider-azurerm/pull/21657 fixes this issue in 3.55.0. I can not test since now I am hitting #21663 :(

**Comment by jvassbo** [comment]
> I think #21657 fixes this issue in 3.55.0. I can not test since now I am hitting #21663 :(

Can confirm issue no longer present in 3.55.0

**Comment by catriona-m** [comment]
It looks like this issue has been resolved by #21657 so I am going to close this now. Thanks!
